site/pages/api/pledge/index.ts

```diff
@@ -0,0 +1,32 @@
+import { NextApiRequest, NextApiResponse } from "next/types";
+import { getPledgeByAddress, pledgeResolver } from "lib/moralis";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
```

Comment:
let's make sure we type the expected response data and request body. We can share the same types on the client too.

Here is some example boilerplate to make this easier, at least for inferring the correct return type:
```ts
const data = await fetchFromEndpoint<PledgeResponse>("/api/pledge", {
    body: JSON.stringify(body),
});

export const fetchFromEndpoint = async <T = any>(
  url: Endpoint,
  init?: RequestInit | undefined
): Promise<T> => {
  try {
    const res = await fetch(`${BASE_URL}${url}`, init);
    if (!res.ok) {
      const json = await res.json();
      throw json;
    }
    const json: T = await res.json();
    return json;
  } catch (e) {
    console.error(e);
    throw e;
  }
};
```